xbmc/addons/binary-addons/BinaryAddonManager.cpp

```diff
@@ -55,6 +57,20 @@ bool CBinaryAddonManager::Init()
   for (auto addon : binaryAddonList)
     AddAddonBaseEntry(addon);
 
+  CServiceBroker::GetAddonMgr().RegisterAddonMgrCallback(ADDON_ADSPDLL, this);
```

Comment:
What's the point here registering for every type? The callback is stored in a map so there's only one possible listener for an addon type so why not just register all types in one method call?